dinghy-lib/src/apple/device.rs

```diff
@@ -601,6 +606,8 @@ fn launch_app(dev: &AppleSimDevice, app_args: &[&str], _envs: &[&str]) -> Result
         .lines()
         .rev()
         .find(|line| line.contains("exited with status"));
+    // This will fail if criterion didn't run.
+    let _ = copy_recursively(criterion_path, "./target/criterion");
```

Comment:
This looks pretty cool. I do wonder why the coping is only happening one way? It seems like we want to send the target/criterion directories contents to the device before running, then let it modify them, then copy them back to target/criterion.

Don't we need different criterion target directories per target or really per device? If I'm benchmarking 3 on 3 models of iPhones and 3 models of Apple Watches, I don't want those benchmark results to be mixed. Maybe I'm missing something though?